### ISSUE TITLE
feature request: add "easy" API to use cfn-lint as a library

### ISSUE THREAD

**Issue by evilensky** [aws-cloudformation/cfn-lint#2243]
Note: I use the term "easy" api as inspired by the [libcurl-easy](https://curl.se/libcurl/c/libcurl-easy.html) API which performs most setup steps automatically for the caller. Without this easy API the user must manually initialize all options required.

This issue is a feature request to add a "simple" or "easy" API to cfn-lint so that it can be used as a library within other applications. The goal would be that without any extra options set, one could call a hypothetical `cfnlint.simple.run()` on a dict or alternatively bytes-like object and return a list of lint violations. There would be a simple way to set options like ignoring rules.

Having the easy api as part of the project will help all users avoid re-inventing this core functionality.

Context:

Our pipeline generates CloudFormation templates via Troposphere as part of a suite of test fixtures. Each fixture could benefit from a `cfn-lint` step. Today these fixtures are generated dynamically and `assert` statements are called for each fixture. An `easy` api would would allow us to set up a parameterized test for linting as well. Without round-tripping the fixture to a file and without the tremendous overhead of calling cfn-lint via `subprocess`.

Alternatively if there are any hints on how we should implement this ourselves, that would be useful too. I am a bit lost: I would like to use as high level functions as possible but there is a `loads()` call made implicitly within `cfnlint.core.get_template_rules()`. Maybe I monkeypatch this to return the template dict I already have?

**Comment by LaurenceWarne** [comment]
Hi, https://github.com/aws-cloudformation/cfn-lint/pull/2285 has now been merged, does https://github.com/aws-cloudformation/cfn-lint/blob/main/docs/getting_started/integration.md#getting-started fit your use case?

**Comment by evilensky** [comment]
@LaurenceWarne That looks really good, thank you so much!  FWIW I eventually fumbled my way to:

```Python
def get_cfn_lint_matches(test_name, test_input):

    try:
        template = cfnlint.decode.cfn_yaml.load(condenser_output)
    except Exception as e:  # pylint: disable=broad-except
        LOG.error("Skipping cfn-lint validation due to an internal error.\n")
        LOG.error(str(e))
        return []

    # Initialize the ruleset to be applied (some excludes)
    # FIXME initialize .cfnlintrc.yaml correctly.
    rules = cfnlint.core.get_rules([], ["W3011", "W3005"], [], [], False, [])

    # Default janus region
    regions = ["us-west-2"]

    # Runs Warning and Error rules
    matches = cfnlint.core.run_checks(test_name, template, rules, regions)

    return matches
```

**Comment by kddejong** [comment]
@evilensky should have this officially released in pypi this week
